pkgs/tools/system/lanzaboote/tool.nix

```diff
@@ -0,0 +1,102 @@
+{ systemd
+, makeWrapper
+, stdenv
+, binutils-unwrapped
+, sbsigntool
+, makeRustPlatform
+, rustPlatform
+, rustc
+, cargo
+, fetchFromGitHub
+, lanzaboote-stub
+, lib
+, mkStdenvNoLibs
+, pkgsBuildTarget
+, buildPackages
+, overrideCC
+, writeShellScriptBin
+}:
+let
+  uefiPlatform = lib.systems.elaborate "${stdenv.hostPlatform.qemuArch}-windows" // {
+    rustc.config = "${stdenv.hostPlatform.qemuArch}-unknown-uefi";
+    useLLVM = true;
+  };
+  uefiLinker = writeShellScriptBin "lld-link" ''
+    exec ${buildPackages.llvmPackages_15.lld}/bin/lld-link "$@"
+    '';
+
+  uefiBintools = buildPackages.llvmPackages_15.clang.bintools.override {
+    extraBuildCommands = ''
+      wrap lld-link ${../../../build-support/bintools-wrapper/ld-wrapper.sh} ${uefiLinker}/bin/lld-link
+    '';
+  };
+
+  uefiCC = buildPackages.llvmPackages_15.clang; #.override {
+ #   bintools = uefiBintools;
+ # };
+
+  hostUefiStdenv = overrideCC (stdenv.override (old: {
```

Comment:
Here, I'm kinda annoyed and I don't know how to deal with it, I discussed it quickly with @Mic92 IRL.

The lanzaboote stub is a binary that is made to run in a UEFI environment, that is lanzaboote stub's hostSystem = UEFI triple of that architecture.

The problem with that is that the tool needs to build it as **data** (not executable code on the tool's host platform which is a Linux environment.), I am not sure what is the proper way or semantic to say "I want to build this that is for another host system as data please", this sounds like cross compilation with build=Linux environment, host=target=UEFI.

The problem is that I need a rustc with build=host, target=UEFI and because rustc is not yet able to compile std for more than its host target, it means, I need to build a fake host uefi stdenv to make rustc understand that I want an extra target that is uefi.

Then I need to plug this rustc variant into a `uefiRustPlatform` with a `uefiStdenv` where build=Linux environment and hostPlatform=targetPlatform.

Does it make sense? Is there a way to simplify it further? Is the only proper fix to refactor the `rustc.nix` process to support `extraTargets` or split up stage0 (rustc binary) → stage1 (host compiler) → stage2 (host compiler), stage 2 (host std), stage 2 (std of a target X) → rustc drv wrapper that assembles a subset of all of this by copying stuff.